### PR TITLE
Improve auc roc documentation and add dash app exploration of docking results

### PR DESCRIPTION
**Pull request by apayne97**
## Description
This was an initial branch to play around with some visualization. Needs to be closed now but it has some scripts I want to keep!
not done things moved to #79 

## Todos
Notable points that this PR has either accomplished or will accomplish.
### Code Organization and Documentation
  - [x] split up plotting, basic analysis, and AUC-ROC analysis into different functionalities
  - [x] improve documentation for all analysis code
 
### Plotting
  - [x] add dash exploration of docking results

## Status
- [x] Ready to go

### CODE REVIEW

File: scripts/dash-all-results-exploration.py [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+                        """
+                        ),
+                        html.Pre(id="click-data", style=styles["pre"]),
+                    ],
+                    className="three columns",
+                ),
+            ],
+        ),
+    ]
+)
```
Comment (ijpulidos) [inline scripts/dash-all-results-exploration.py:224]: We probably want to have this in a separate file that we can just import, such that this script is easier to read and handle, when needed.

File: scripts/dash-all-results-exploration.py [hunk 2]
```diff
@@ -0,0 +1,589 @@
+import pandas as pd
```
Comment (ijpulidos) [inline scripts/dash-all-results-exploration.py:1]: We want a docstring here telling potential users and maintainers what this script does, with inputs and outputs. In a few words.

File: scripts/dash-docking-exploration.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
+import pandas as pd
```
Comment (ijpulidos) [inline scripts/dash-docking-exploration.py:1]: We want a docstring here telling potential users and maintainers what this script does, with inputs and outputs. In a few words.

File: scripts/dash-docking-exploration.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            ],
+            style={
+                # "width": "49%",
+                "display": "inline-block",
+                # "float": "right",
+                "padding": "0 20",
+            },
+        ),
+    ]
+)
```
Comment (ijpulidos) [inline scripts/dash-docking-exploration.py:134]: How is this different from the similar code in the other script?

We probably want to have this in a separate file that we can just import, such that this script is easier to read and handle, when needed. And reuse it that way, if possible.

File: scripts/dash-interactive-graphs.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
+from dash import Dash, html, dcc, Input, Output
```
Comment (ijpulidos) [inline scripts/dash-interactive-graphs.py:1]: We want a docstring here telling potential users and maintainers what this script does, with inputs and outputs. In a few words.

File: scripts/dash-scatterplot.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
+from dash import Dash, dcc, html, Input, Output
```
Comment (ijpulidos) [inline scripts/dash-scatterplot.py:1]: We want a docstring here telling potential users and maintainers what this script does, with inputs and outputs. In a few words.

**Review by ijpulidos — COMMENTED** [review]
A few comments, I don't know how critical is the code in this branch, though.

**Comment by ijpulidos** [comment]
Ah, also as a general remark I forgot to say that we probably want to make sure the Jupyter notebooks only have the input cells, that is, for then not to show the outputs (unless needed), such that the changes in the git history do not explode.

**Comment by apayne97** [comment]
> Ah, also as a general remark I forgot to say that we probably want to make sure the Jupyter notebooks only have the input cells, that is, for then not to show the outputs (unless needed), such that the changes in the git history do not explode.

I've cleared the outputs in this branch and will try to remember to do that for all the other branches going forward! that's a good idea.

File: scripts/analyze_docking_results.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import sys, os, argparse
```
[the remark this answers is not in the document]
Comment (apayne97) [reply scripts/analyze_docking_results.py]: actually, this script is basically a copy and has now been deleted.

File: scripts/dash-all-results-exploration.py (hunk 1 above)
Comment (apayne97) [reply scripts/dash-all-results-exploration.py:219]: I agree! this is beyond me right now though

File: scripts/dash-interactive-graphs.py (hunk 5 above)
Comment (apayne97) [reply scripts/dash-interactive-graphs.py:1]: file deleted

File: scripts/dash-scatterplot.py (hunk 6 above)
Comment (apayne97) [reply scripts/dash-scatterplot.py:1]: file delete! just an example

**Review by ijpulidos — APPROVED** [review]
Nice, over 50k lines less, that's what I wanted to see by removing the outputs of the notebooks.

I know the code here is experimental, and if we don't really plan to use this in production anytime soon, this should be good to be merged and improved upon. Hence, the comments are non-blocking.

**Pull request merged**